### PR TITLE
ci: job to check contracts build warnings

### PR DESCRIPTION
**Pull request by smartcontracts**
Modifies the current CI workflow to check for compiler warnings.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2141,7 +2137,10 @@ workflows:
       - cannon-prestate:
           requires:
             - go-mod-download
-      - contracts-bedrock-build
+      - contracts-bedrock-build:
+          build_command: forge build --skip test
+          context:
+            - slack
```
Comment (ajsutton) [inline .circleci/config.yml:2143]: Seems like we should be verifying the contracts build before the PR can merge so this shouldn't ever fail.  Having to specify custom build commands feels like it's adding duplication and is going to get out of sync so we aren't necessarily testing contracts built the same was as we would for a release here. Might be more maintainable for `contracts-bedrock-build` to just accept a `skip_tests` param and it always controls the build command.

**Review by ajsutton — APPROVED** [review]
LGTM but would be good to follow up and make the parameters to contracts-bedrock-build more declarative rather than having custom build commands duplicated around the config which are likely to get out of sync.

File: .circleci/config.yml (hunk 1 above)
Comment (smartcontracts) [reply .circleci/config.yml:2143]: Agreed, am working to get there.

Comment (smartcontracts) [reply .circleci/config.yml:2143]: Recent foundry feature that adds junit test report files will make this possible

**Pull request merged**